libs/sysplugins/smarty_internal_data.php

```diff
@@ -96,6 +96,10 @@ public function __construct()
      */
     public function assign($tpl_var, $value = null, $nocache = false)
     {
+        if(is_array($tpl_var) && $value !== null) {
```

Comment:
I would change the test to 
```suggestion
        if(is_array($tpl_var) && isset($value)) {
```
While semantically the same, `isset()` conveys the meaning of the test more clearly.